### PR TITLE
Set notehead scheme from XML `<notehead>` tag

### PR DESCRIPTION
**Pull request by miiizen**
This sets a note's `NoteHeadScheme` to `HEAD_PITCHNAME` when the value of the `<notehead>` tag is "named".

Screenshot 2024-04-25 at 16 22 15

### CODE REVIEW

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 1]
```diff
@@ -6146,6 +6149,7 @@ Note* MusicXMLParserPass2::note(const String& partId,
         handleSmallness(cue || isSmall, note, c);
         note->setPlay(!cue);          // cue notes don't play
         note->setHeadGroup(headGroup);
+        note->setHeadScheme(headScheme);
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:6152]: Better only set it `if (headScheme != NoteHeadScheme::HEAD_NORMAL)`

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 2]
```diff
@@ -5957,6 +5958,8 @@ Note* MusicXMLParserPass2::note(const String& partId,
             auto noteheadValue = m_e.readText();
             if (noteheadValue == "none") {
                 hasHead = false;
+            } else if (noteheadValue == "named") {
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:5961]: Such a notehead type doesn't exist in MusicXML, see https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/notehead/ and https://www.w3.org/2021/06/musicxml40/musicxml-reference/data-types/notehead-value/
Esp. this:
Noteheads in the SMuFL [Note name noteheads](https://www.w3.org/2021/03/smufl14/tables/note-name-noteheads.html) and [Note name noteheads supplement](https://www.w3.org/2021/03/smufl14/tables/note-name-noteheads-supplement.html) ranges (U+E150–U+E1AF and U+EEE0–U+EEFF) should not use the smufl attribute or the other value, but instead use the <notehead-text> element.

File: src/importexport/musicxml/tests/data/testNamedNoteheads.xml [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+        </attributes>
+      <note default-x="110.93" default-y="-45">
+        <pitch>
+          <step>D</step>
+          <octave>4</octave>
+          </pitch>
+        <duration>1</duration>
+        <voice>1</voice>
+        <type>quarter</type>
+        <notehead>named</notehead>
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/tests/data/testNamedNoteheads.xml:122]: Results in:
Fatal error: line 122 column 18 Content of element notehead does not match its type definition: String content is not listed in the enumeration facet..

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp (hunk 2 above)
Comment (miiizen) [reply src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:5961]: Apologies, this is an out of spec case for a single exporter

Comment (Jojo-Schmitz) [reply src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp]: Ah, I see, thanks

**Review by oktophonie — APPROVED** [review]

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**